crates/matrix-sdk-crypto/src/types/cross_signing/master.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{collections::btree_map::Iter, sync::Arc};
+
+use ruma::{encryption::KeyUsage, DeviceKeyId, OwnedDeviceKeyId, UserId};
+use serde::{Deserialize, Serialize};
+use vodozemac::Ed25519PublicKey;
+
+use super::{CrossSigningKey, CrossSigningSubKeys, SigningKey};
+use crate::{
+    olm::VerifyJson,
+    types::{Signatures, SigningKeys},
+    SignatureError,
+};
+
+/// Wrapper for a cross signing key marking it as the master key.
+///
+/// Master keys are used to sign other cross signing keys, the self signing and
+/// user signing keys of an user will be signed by their master key.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(try_from = "CrossSigningKey")]
+pub struct MasterPubkey(pub(super) Arc<CrossSigningKey>);
```

Comment:
It seems to have been moved from `crates/matrix-sdk-crypto/src/identities/user.rs`, except that the `PartialEq` impl' has been removed (it's now in the `mod.rs` module).

I'm not checking the implementations, it seems to be a “simple” move. Can you confirm please?

Reply:
Yep, it's just a move. We have now comprehensive tests for the `PartialEq` implementation so we don't mess things up by accident.